### PR TITLE
Split regression.uts in multiple smaller files

### PR DESCRIPTION
**Pull request by polybassa**
As discussed in #2550. This PR splits the regression.uts file into multiple smaller parts.

### CODE REVIEW

File: test/regression/network.uts [hunk 1]
```diff
@@ -136,20 +136,20 @@ retry_test(_test)
 = AS resolvers
 ~ netaccess IP as_resolvers
 * This test retries on failure because it often fails
-
+# TODO: Fix this test
```
Comment (polybassa) [inline test/regression/network.uts:139]: @gpotter2 Could you please help me to fix this test?

File: test/regression/tcp_ip.uts [hunk 2]
```diff
@@ -295,25 +294,25 @@ p = DNS(raw(DNS(id=1,ra=1,an=DNSRR(rrname='secdev', type='TXT', rdata=["sweet",
 assert p[DNS].an.rdata == [b"sweet", b"celestia"]
 assert raw(p) == b'\x00\x01\x01\x80\x00\x00\x00\x01\x00\x00\x00\x00\x06secdev\x00\x00\x10\x00\x01\x00\x00\x00\x01\x00\x0f\x05sweet\x08celestia'
 
-= DNS - Malformed DNS over TCP message
+#= DNS - Malformed DNS over TCP message
```
Comment (polybassa) [inline test/regression/tcp_ip.uts:297]: @gpotter2 Could you please help me to fix this test?

**Comment by gpotter2** [comment]
Thanks a lot for the PR. I don't think you need to put the files under `regressions/`. (this file dates back from when all of scapy's unit testing was done in a single "regression" file :smile:)

**Comment by polybassa** [comment]
Alright, I'll move it. I put it there to better keep track of the files.
--

File: test/regression/tcp_ip.uts [hunk 3]
```diff
@@ -295,25 +294,25 @@ p = DNS(raw(DNS(id=1,ra=1,an=DNSRR(rrname='secdev', type='TXT', rdata=["sweet",
...
+#except TypeError as e:
+#    warning("What's going on here?")
+#    assert True
+
+#try:
+#    raise Scapy_Exception("Malformed DNS message: invalid length!")
+#    p = IP(raw(IP()/TCP()/DNS(length=28, qdcount=1)))
+#    assert False
+#except Scapy_Exception as e:
+#    assert str(e) == "Malformed DNS message: invalid length!"
```
Comment (gpotter2) [inline test/regression/tcp_ip.uts:315]:
```suggestion
old_d = conf.debug_dissesctor
conf.debug_dissector = True
try:
    inner = IP()/TCP()/DNS(length=28)
    p = IP(raw(inner)[:-13])
    assert False
except Scapy_Exception as e:
    assert str(e) == "Malformed DNS message: too small!"
except TypeError as e:
    warning("What's going on here?")
    assert True

try:
    p = IP(raw(IP()/TCP()/DNS(length=28, qdcount=1)))
    assert False
except Scapy_Exception as e:
    assert str(e) == "Malformed DNS message: invalid length!"
conf.debug_dissector = old_d
```

**Comment by polybassa** [comment]
@guedou
Thanks for your feedback. Your points are completely reasonable. 

I will not continue on this branch, but maybe it's useful for a piece by piece fragmentation of regression.uts. 

Also during this split, to had the feeling, that some tests could be combined according to there test purpose. But it was pretty hard for me to see the logical relationships. This is something, you are probably much better in. 

I will pick a set of tests and make an example of what you suggested, to obtain another case for my pytest branch.

**Comment by polybassa** [comment]
I don't think this is useful anymore.

**Pull request closed without merging**